macros/src/types/enum.rs

```diff
@@ -95,12 +95,26 @@ fn format_variant(
         },
         Tagged::Adjacently { tag, content } => match &variant.fields {
             Fields::Unnamed(unnamed) if unnamed.unnamed.len() == 1 => {
-                let FieldAttr { type_override, .. } =
-                    FieldAttr::from_attrs(&unnamed.unnamed[0].attrs)?;
+                let FieldAttr {
+                    type_as,
+                    type_override,
+                    ..
+                } = FieldAttr::from_attrs(&unnamed.unnamed[0].attrs)?;
+
+                if let (Some(_type_as), Some(_type_override)) = (&type_as, &type_override) {
+                    syn_err!("`type` is not compatible with `as`")
+                }
+
+                let parsed_ty = if let Some(_type_as) = &type_as {
```

Comment:
Rename the variable to remove the `_` prefix, and if possible use `ref` instead of `&`